### PR TITLE
install CNI on linux

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:

Fixing the CI

#### Which issue(s) this PR fixes:

See conversation here: https://github.com/kubernetes-sigs/cri-tools/pull/914#issuecomment-1101812540

#### Special notes for your reviewer:

I'm not sure how to test it though. Let's see if CI will pick it up

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
ok, the step in question seems to be working:

```
Run sudo -E PATH=$PATH script/setup/install-cni $(grep containernetworking/plugins go.mod | awk '{print $[2](https://github.com/kubernetes-sigs/cri-tools/runs/6069479239?check_suite_focus=true#step:18:2)}')
  sudo -E PATH=$PATH script/setup/install-cni $(grep containernetworking/plugins go.mod | awk '{print $2}')
  shell: /bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    GOROOT: /opt/hostedtoolcache/go/1.17.8/x6[4](https://github.com/kubernetes-sigs/cri-tools/runs/6069479239?check_suite_focus=true#step:18:4)
    GOPATH: /home/runner/work/cri-tools/cri-tools
Cloning into '/tmp/tmp.uEkOsnitzV/plugins'...
/tmp/tmp.uEkOsnitzV/plugins ~/work/cri-tools/cri-tools/src/github.com/containerd/containerd
Note: switching to 'v1.1.1'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 4744ec2 Merge pull request #71[6](https://github.com/kubernetes-sigs/cri-tools/runs/6069479239?check_suite_focus=true#step:18:6) from squeed/cp-bugfixes
Building plugins 
  bandwidth
  firewall
  portmap
  sbr
  tuning
  vrf
  bridge
  host-device
  ipvlan
  loopback
  macvlan
  ptp
  vlan
  dhcp
  host-local
  static
{
  "cniVersion": "1.0.0",
  "name": "containerd-net",
  "plugins": [
    {
      "type": "bridge",
      "bridge": "cni0",
      "isGateway": true,
      "ipMasq": true,
      "promiscMode": true,
      "ipam": {
        "type": "host-local",
        "ranges": [
          [{
            "subnet": "10.[8](https://github.com/kubernetes-sigs/cri-tools/runs/6069479239?check_suite_focus=true#step:18:8)8.0.0/[16](https://github.com/kubernetes-sigs/cri-tools/runs/6069479239?check_suite_focus=true#step:18:16)"
          }],
          [{
            "subnet": "[20](https://github.com/kubernetes-sigs/cri-tools/runs/6069479239?check_suite_focus=true#step:18:20)01:[48](https://github.com/kubernetes-sigs/cri-tools/runs/6069479239?check_suite_focus=true#step:18:48)[60](https://github.com/kubernetes-sigs/cri-tools/runs/6069479239?check_suite_focus=true#step:18:60):4860::/[64](https://github.com/kubernetes-sigs/cri-tools/runs/6069479239?check_suite_focus=true#step:18:64)"
          }]
        ],
        "routes": [
          { "dst": "0.0.0.0/0" },
          { "dst": "::/0" }
        ]
      }
    },
    {
      "type": "portmap",
      "capabilities": {"portMappings": true}
    }
  ]
}
~/work/cri-tools/cri-tools/src/github.com/containerd/containerd
```

File: .github/workflows/containerd.yml [hunk 1]
```diff
@@ -169,6 +169,13 @@ jobs:
           ginkgo version
           cp $(command -v ginkgo) /usr/local/bin
 
+      - name: Install cni on Linux
+        if: startsWith(matrix.os, 'ubuntu')
+        shell: bash
+        run: |
+          sudo -E PATH=$PATH script/setup/install-cni $(grep containernetworking/plugins go.mod | awk '{print $2}')
```
Comment (mikebrow) [inline .github/workflows/containerd.yml:176]: might want to move this up to line 101 and do it right after installing containerd node working directory  
`${{ github.workspace }}/src/github.com/containerd/containerd`

**Review by mikebrow — COMMENTED** [review]
see comment

**Comment by SergeyKanzhelev** [comment]
Interestingly some seemingly CNI related tests failed:

```
• Failure [62.644 seconds]
[k8s.io] Streaming
github.com/kubernetes-sigs/cri-tools/pkg/framework/framework.go:72
  runtime should support streaming interfaces
  github.com/kubernetes-sigs/cri-tools/pkg/validate/streaming.go:55
    runtime should support portforward [Conformance] [It]
    github.com/kubernetes-sigs/cri-tools/pkg/validate/streaming.go:124

    Timed out after 60.000s.
    Expected
        <*url.Error | 0xc0000a3320>: {
            Op: "Get",
            URL: "http://127.0.0.1:12001/",
            Err: <*net.OpError | 0xc00007ee10>{
                Op: "dial",
                Net: "tcp",
                Source: nil,
                Addr: <*net.TCPAddr | 0xc0000a3170>{
                    IP: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 127, 0, 0, 1],
                    Port: 12001,
                    Zone: "",
                },
                Err: <*os.SyscallError | 0xc0005c24e0>{
                    Syscall: "connect",
                    Err: <syscall.Errno>0x6f,
                },
            },
        }
    to be nil

    github.com/kubernetes-sigs/cri-tools/pkg/validate/networking.go:253
```

File: .github/workflows/containerd.yml (hunk 1 above)
Comment (SergeyKanzhelev) [reply .github/workflows/containerd.yml:176]: I wanted to keep is consistent with Windows. Perhaps both CNI installations can be moved to earlier stage - right after containerd installation

Comment (mikebrow) [reply .github/workflows/containerd.yml:116]: kk..

**Comment by SergeyKanzhelev** [comment]
> Interestingly some seemingly CNI related tests failed:
> 
> ```
> • Failure [62.644 seconds]
> [k8s.io] Streaming
> github.com/kubernetes-sigs/cri-tools/pkg/framework/framework.go:72
>   runtime should support streaming interfaces
>   github.com/kubernetes-sigs/cri-tools/pkg/validate/streaming.go:55
>     runtime should support portforward [Conformance] [It]
>     github.com/kubernetes-sigs/cri-tools/pkg/validate/streaming.go:124
> 
>     Timed out after 60.000s.
>     Expected
>         <*url.Error | 0xc0000a3320>: {
>             Op: "Get",
>             URL: "http://127.0.0.1:12001/",
>             Err: <*net.OpError | 0xc00007ee10>{
>                 Op: "dial",
>                 Net: "tcp",
>                 Source: nil,
>                 Addr: <*net.TCPAddr | 0xc0000a3170>{
>                     IP: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 127, 0, 0, 1],
>                     Port: 12001,
>                     Zone: "",
>                 },
>                 Err: <*os.SyscallError | 0xc0005c24e0>{
>                     Syscall: "connect",
>                     Err: <syscall.Errno>0x6f,
>                 },
>             },
>         }
>     to be nil
> 
>     github.com/kubernetes-sigs/cri-tools/pkg/validate/networking.go:253
> ```

ok, this might have been a flake. We can ignore fore now I think

**Review by mikebrow — APPROVED** [review]
LGTM
non binding

**Comment by mikebrow** [comment]
> Interestingly some seemingly CNI related tests failed:
> 
> ```
> • Failure [62.644 seconds]
> [k8s.io] Streaming
> github.com/kubernetes-sigs/cri-tools/pkg/framework/framework.go:72
>   runtime should support streaming interfaces
>   github.com/kubernetes-sigs/cri-tools/pkg/validate/streaming.go:55
>     runtime should support portforward [Conformance] [It]
>     github.com/kubernetes-sigs/cri-tools/pkg/validate/streaming.go:124
> 
>     Timed out after 60.000s.
>     Expected
>         <*url.Error | 0xc0000a3320>: {
>             Op: "Get",
>             URL: "http://127.0.0.1:12001/",
>             Err: <*net.OpError | 0xc00007ee10>{
>                 Op: "dial",
>                 Net: "tcp",
>                 Source: nil,
>                 Addr: <*net.TCPAddr | 0xc0000a3170>{
>                     IP: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 127, 0, 0, 1],
>                     Port: 12001,
>                     Zone: "",
>                 },
>                 Err: <*os.SyscallError | 0xc0005c24e0>{
>                     Syscall: "connect",
>                     Err: <syscall.Errno>0x6f,
>                 },
>             },
>         }
>     to be nil
> 
>     github.com/kubernetes-sigs/cri-tools/pkg/validate/networking.go:253
> ```

yeah saw that will try to repro it locally easy enough to run it a few thousand times. :-)

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**